presto-main/src/main/java/com/facebook/presto/operator/index/DynamicTupleFilterFactory.java

```diff
@@ -55,7 +55,7 @@ public OperatorFactory filterWithTuple(Page tuplePage)
     {
         Page normalizedTuplePage = normalizeTuplePage(tuplePage);
         TupleFilterProcessor processor = new TupleFilterProcessor(normalizedTuplePage, outputTypes, outputFilterChannels);
-        return new FilterAndProjectOperatorFactory(filterOperatorId, planNodeId, processor, outputTypes);
+        return new FilterAndProjectOperatorFactory(filterOperatorId, planNodeId, () -> processor, outputTypes);
```

Comment:
I would move the processor creation into a lambda so that you get a new instance per operator created... It maybe that this is ok for now, but someone could make a change in the future that assumes the single instance per operator model.